PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+
+## Parcel CLI
+
+The Parcel CLI is built into the main `parcel` package. While you can install
+it globally and run it, it is much better to install it locally into your
+project as a dev dependency.
+
+```sh
+yarn add --dev parcel
+```
+
+You should also add some "scripts" to your `package.json` to run it easier.
+
+```json
+{
+  "name": "my-project",
+  "scripts": {
+    "build": "parcel --production",
+    "start": "parcel --serve"
+  },
+  "devDependencies": {
+    "parcel": "latest"
+  }
+}
+```
+
+Now you can run `yarn build` to bundle your project for production and
+`yarn start` to dev on your project.
+
+### CLI Args & Flags
+
+Usage:
+
+```sh
+$ parcel [...entries] [...flags]
+```
+
+#### `[...entries]`
+
+Entry files to start bundling, these will be preserved as entry points in the
+output. Defaults to `package.json#source`, falling back to `src/index.*` or
+`index.*`. See [#Entries](#entries-sources-targets-environment).
+
+#### `--serve, -s`
+
+Serve assets on a local server
+
+#### `--watch, -w`
+
+Watch and rebuild code on file changes.
+
+#### `--open, -o [browser]`
+
+Open your local server in a browser. You can optionally pass the name of the
+browser you want to open, otherwise it will use your default browser.
+
+#### `--port <port>`
+
+Configure the port to serve assets on. Alternatively you can use the `$PORT`
+environment variable.
+
+#### `--https`
+
+This will generate a local certificate (which will be untrusted by your
+browser, you'll need to approve it) and serve your assets over `https://`
+
+##### `--cert <path>`
+
+Specify the filepath to your SSL certificate when using `--https`.
+
+##### `--key <path>`
+
+Specify the filepath to your SSL key when using `--https`.
+
+#### `--cache <dir>`, `--no-cache`
+
+Configure the cache directory with `--cache <dir>` or disable it altogether
+with `--no-cache`.
+
+#### `--hot`, `--no-hot`
+
+Turn hot reloading on or off.
+
+##### `--hot-hostname <hostname>`
+
+Configure the hot reloading hostname.
+
+##### `--hot-port <port>`
+
+Configure the hot reloading port.
+
+#### `--[no-]source-maps`
+
+Turn source maps on or off. Source maps are turned on by default except in
+production builds.
+
+#### `--autoinstall [npm/yarn], --no-autoinstall`
+
+When enabled, whenever Parcel discovers a dependency that isn't installed it
+will attempt to install it with either npm or Yarn (defaults to npm unless a
+`yarn.lock` exists).
+
+#### `--mode <mode>`
+
+Override the environment mode, including those manually configured to something
+else (ex. `--mode production` or `--mode development`). Defaults to the
+`NODE_ENV` environment variable.
+
+##### `--development, --dev`
+
+Aliases for `--mode development`.
+
+##### `--production, --prod`
+
+Aliases for `--mode production`.
+
+##### `--test`
+
+Aliases for `--mode test`.
+
+#### `--public-url <url>`
+
+[todo]
+
+#### `--log-level <level>`
+
+Set the log level, either "0" (no output), "1" (errors), "2" (warnings +
```

Comment:
It would read cleaner to list these out:

Sets the log level (Default 2):
1. `0`: No output
1. `1` : Errors
1. `2`: Warnings and Errors
1. `3`: All